src/libspats/util.cpp

```
@@ -0,0 +1,249 @@
+#include "util.h"
+
+namespace spats {
+
+using namespace secp_primitives;
+
+// Encrypt a diversifier using AES-256
+std::vector<unsigned char> SpatsUtils::diversifier_encrypt(const std::vector<unsigned char>& key, const uint64_t i) {
+    // Serialize the diversifier
+    CDataStream i_stream(SER_NETWORK, PROTOCOL_VERSION);
+    i_stream << i;
+
```

cppcheck (2.10-2)

[performance] Function parameter 'label' should be passed by const reference.

(passedByValue)